src/main/webapp/app/markdown-editor/commands/command.ts

```diff
@@ -79,4 +81,35 @@ export abstract class Command {
     }
 
     abstract execute(input?: string): void;
+
+    protected ignoreWhiteSpace(text: string) {
```

Comment:
- What does this method actually do? If it only removes whitespace in the beginning and end you can use String.prototype.trim, e.g. `text.trim()`.
- A logic heavy method like this needs documentation
- I would avoid to both return a value in a method and change the state of the object (here this.startSpace/this.endspace is set to the identifier of the command). The method name is also misleading here.

I would propose something different:
- startSpace and endSpace should be initialized by the respective command (e.g. italic)
- these identifiers can then be used to check if the selected text is within the command identifiers
- Trimming can be done with a call to `text.trim()`

Reply:
Unfortunately `text.trim()` won't do the job here since we need to reinsert the white-spaces after adding the markdowns. So my method counts the white-space characters at the beginning and end of selected text and saves them in other strings (startSapce and endSpace), ignore the white-spaces until markdowns are added around actual text then reinserts the white space.
Example: if the selected text is "I  am  testing" and we select "  am  " using `text.trim()` will return "I**am**testing" which is not the expected behavior.
I agree with you that it is a logic heavy method for what it actually solves.
I noticed that the markdown-editor already deletes more than one successive white-spaces if you click on 'preview' (see screenshot) so another solution would be to use `text.trim()` and only add one white-space per default. In this case we will have about 3 if cases to check where to insert the white-space (at the beginning,end,both or non)